src/mesh/mesh_generator.jl

```diff
@@ -364,6 +367,104 @@ function _rectangle_quad_mesh(nx, ny, xmin, xmax, ymin, ymax, ::Val{2}, bnd_name
     return Mesh(nodes, celltypes, Connectivity(cell2nnodes, cell2node))
 end
 
+function hexa_mesh(
+    nx,
+    ny,
+    nz;
+    xmin = 0.0,
+    xmax = 1.0,
+    ymin = 0.0,
+    ymax = 1.0,
+    zmin = 0.0,
+    zmax = 1.0,
+    order = 1,
+    bnd_names = ("xmin", "xmax", "ymin", "ymax", "zmin", "zmax"),
+)
+    @assert order == 1 "Not implemented for order = $order"
+
+    lx = xmax - xmin
+    ly = ymax - ymin
+    lz = zmax - zmin
+    nelts = (nx - 1) * (ny - 1) * (nz - 1)
+    Δx = lx / (nx - 1)
+    Δy = ly / (ny - 1)
+    Δz = lz / (nz - 1)
+
+    # Prepare boundary nodes
+    tag2name = Dict(tag => name for (tag, name) in enumerate(bnd_names))
+    tag2nodes = Dict(tag => Int[] for tag in 1:length(bnd_names))
+
+    # Nodes
+    iglob = 1
+    nodes = Array{Node{3, Float64}}(undef, nx * ny * nz)
+    for iz in 1:nz
+        for iy in 1:ny
+            for ix in 1:nx
+                nodes[(iz - 1) * nx * ny + (iy - 1) * nx + ix] =
+                    Node([xmin + (ix - 1) * Δx, ymin + (iy - 1) * Δy, zmin + (iz - 1) * Δz])
+
+                # Boundary conditions
+                if ix == 1
+                    push!(tag2nodes[1], iglob)
+                elseif ix == nx
+                    push!(tag2nodes[2], iglob)
+                elseif iy == 1
+                    push!(tag2nodes[3], iglob)
+                elseif iy == ny
+                    push!(tag2nodes[4], iglob)
+                elseif iz == 1
+                    push!(tag2nodes[5], iglob)
+                elseif iz == ny
+                    push!(tag2nodes[6], iglob)
+                end
+
+                iglob += 1
+            end
+        end
+    end
+
+    # Cell -> node connectivity
+    cell2node = zeros(Int, 8 * nelts)
+    for iz in 1:(nz - 1)
+        for iy in 1:(ny - 1)
+            for ix in 1:(nx - 1)
+                ielt = (iz - 1) * (nx - 1) * (ny - 1) + (iy - 1) * (nx - 1) + ix
+
+                cell2node[8 * ielt - 7] =
+                    (iz - 1 + 0) * nx * ny + (iy - 1 + 0) * nx + ix + 0
+                cell2node[8 * ielt - 6] =
+                    (iz - 1 + 0) * nx * ny + (iy - 1 + 0) * nx + ix + 1
+                cell2node[8 * ielt - 5] =
+                    (iz - 1 + 0) * nx * ny + (iy - 1 + 1) * nx + ix + 1
+                cell2node[8 * ielt - 4] =
+                    (iz - 1 + 0) * nx * ny + (iy - 1 + 1) * nx + ix + 0
+                cell2node[8 * ielt - 3] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 0) * nx + ix + 0
+                cell2node[8 * ielt - 2] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 0) * nx + ix + 1
+                cell2node[8 * ielt - 1] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 1) * nx + ix + 1
+                cell2node[8 * ielt - 0] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 1) * nx + ix + 0
+            end
+        end
+    end
```

Comment:
could you try the following ? 
```julia
cell2node = zeros(Int, 8, nelts)
ijk_nodes = CartesianIndices((1:nx, 1:ny, 1:nz))
id_nodes = LinearIndices(ijk_nodes)
ijk_cells = CartesianIndices((1:(nx - 1), 1:(ny - 1), 1:(nz - 1)))
id_cells = LinearIndices(ijk_cells)
node_order = SA[1, 2, 4, 3, 5, 6, 8, 7]
stencilcube = CartesianIndices((0:1, 0:1, 0:1))[node_order]
for ic in id_cells
    cell2node[:, ic] .= id_nodes[ijk_cells[ic] .+ stencilcube]
end
cell2node = vec(cell2node)
```

Reply:
I'd rather not to. I thought of it too, but I'd like use to do this in a separate PRs, where we could generate line, rectangle and hexa mesh with the same route (using Linear and Cartesian indexing). But for now I just need this simple hexa mesh for dev purpose so let's improve this later.

Reply:
Ok, I can do that latter